src/app_service/service/message/service.nim

```diff
@@ -526,8 +526,8 @@ QtObject:
       self.finishAsyncSearchMessagesWithError(chatId, "search messages response doesn't contain messages array")
       return
 
-    if (messagesArray.kind != JArray):
-      self.finishAsyncSearchMessagesWithError(chatId, "expected messages json array is not of JArray type")
+    if (messagesArray.kind notin {JArray, JNull}):
```

Comment:
I still don't get why you check for `JNull`. If it is `JNull`, that means that it's **not** a `JArray`, then we go in the condition and exit with `finishAsyncSearchMessagesWithError`. If it **is** `JNull`, we want to exit right?

Reply:
Aaah ok, I didn't know that doing `getElems` on JNull was fine. Thanks for the info.